packages/webcrack/src/ast-utils/matcher.ts

```diff
@@ -195,6 +194,21 @@ export function isReadonlyObject(
   );
 }
 
+export function isConstantBinding(binding: Binding) {
+  // Workaround because sometimes babel treats the VariableDeclarator/binding itself as a violation
+  if (binding.constant || binding.constantViolations[0] === binding.path)
+    return true;
+
+  // if there is only a single assignment to the variable and it is a param
+  // then consider it a constant (this may not be very safe)
+  if (binding.kind === 'param' && binding.constantViolations.length === 1) {
+    const [path] = binding.constantViolations;
+    if (path.isAssignmentExpression()) return true;
+  }
```

Comment:
As per the comment, this may not be safe. Still haven't found a great way to check if a variable was used before initialization.

Reply:
a72c8f738c8adf8a35f136cd453b6ec57ab0748b should've improved this